configure.ac

```diff
@@ -446,7 +446,7 @@ fi
 
 ERROR_CXXFLAGS=
 if test "x$enable_werror" = "xyes"; then
-  if test "x$CXXFLAG_WERROR" = "x"; then
+  if test "x$CXXFLAG_WERROR" = ""; then
```

Comment:
cf03e330f88f8d9b140fd96bba8ba4180c1321d5 makes no sense to me, should drop it

Reply:
bitcoin#20182 doesn't work without this fix

Reply:
well.. you didn't fix anything here, you simply broke the condition because `test "x$CXXFLAG_WERROR" = ""` fails with any `$CXXFLAG_WERROR`